Custom actions

nskartoredjo opened the pull request:
As a developer I want to buy land by giving: 
- the width and depth of the desired land,
- the distance from the road,
- if I want to buy land which contains buildings or not,
- the price per squared meter,
- the zone where the desired land should be bought.

So that the GOAL agent should not worry about the multi-polygon calculations.

```
buy_specified_land(SID, ZoneID, BuildingInd, Price, Depth, Width, DistanceToRoad):

SID: stakeholder ID.
ZoneID: Zone ID.
BuildingInd: A indicator from 0 to 2 which specifies the desired land to:
- 0: exclude building areas.
- 1: only use building areas.
- 2: include building areas.
Price: the price per squared meter.
Depth: The depth of the desired land, if, in addition to the depth, it equals zero, this specification will be skipped.
Width: The width of the desired land  if, in addition to the width, it equals zero, this specification will be skipped.
DistanceToRoad: the distance from the desired land from the road. If zero, this specification will be skipped.

```

nskartoredjo committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into custom-actions

nskartoredjo committed:
add setup sell specified land

nskartoredjo committed:
remove checkstyle check

- removed because of prototype state
- also refactored the class

nskartoredjo committed:
remove all kinds of checks

nskartoredjo committed:
remove the echo prints from travis.yml

nskartoredjo committed:
Add start implementation

- also added bat files for debugging reasons

nskartoredjo committed:
change parameters of map_buy_land

nskartoredjo committed:
add support for depth and width

also including:
- distance to road
- 3 choices for including or excluding buildings.

nskartoredjo committed:
add water detection

- does not get used, does not work in the used project
- also removed bridge from road detection

nskartoredjo committed:
add java docs to the action

jannelie commented on contextvh/src/main/java/contextvh/ContextEntity.java:
```diff
@@ -34,6 +34,7 @@ public ContextEntity(final EntityListener env, final String intendedStakeholder,
 
 	@Override
 	public Percept performAction(final Action action) throws TranslationException {
+		System.out.println(action);
```
Why do you have this print action?

HarmenKroon commented on contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java:
```diff
@@ -0,0 +1,338 @@
+	 * - It retrieves the values.
+	 * 
+	 * - within the parameter list. If there are more parameters, then it will
+	 * return a exception.
```
an exception

HarmenKroon commented on contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java:
```diff
@@ -0,0 +1,338 @@
+			MultiPolygon buyableLand = getBuyableLand(sID.intValue(), zoneID.intValue());
+
+			if (buyOnBuilding.intValue() == 0)
+				buyableLand = excludeBuildingLand(buyableLand);
```
0: exclude buildings
1: include buildings or only buildings?
2: only buildings or include buildings?

nskartoredjo commented:
Also, testing and checkstyle would be added by using new branches from the develop branch, after the merge.

HarmenKroon commented on contextvh/src/main/java/contextvh/actions/BuySpecifiedLand.java:
```diff
@@ -0,0 +1,338 @@
+	 * @param distanceToRoad
+	 *            the distance the new area should be from a road. If zero, this
+	 *            check will be skipped.
+	 * @return
```
Create actionspec with explanation about zeros and skips

HarmenKroon commented:
Why is travis changed?

nskartoredjo commented:
@HarmenKroon it skips the checkstyle and coverage. See previous comment

ardwennem2 commented:
Approved

nskartoredjo committed:
fix problems stated in the PR comments

- remove print
- add additional comment
- fixed spelling
- add 'final' to every method arguments

HarmenKroon commented:
Approved

jannelie commented:
Approved

jannelie commented:
Approved once again for pullapprove 

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/nskartoredjo/tygron/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=nskartoredjo/tygron)

HarmenKroon commented:
Approved

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/nskartoredjo/tygron/pull-request/2/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=nskartoredjo/tygron)
